src/Aspire.Hosting/Utils/ReferenceExpressionParser.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.RegularExpressions;
+
+namespace Aspire.Hosting.Utils;
+internal static partial class ReferenceExpressionParser
+{
+    [GeneratedRegex(@"{(?!\d)")]
+    private static partial Regex LeftNonParameterBraceRegex();
+    [GeneratedRegex(@"(?<!\d)}")]
```

Comment:
How does this handle something that ends in a digit like `myVar1`?

Reply:
It doesn't handle these cases currently. Thinking about potential use cases more, it also doesn't handle valid parameter formatting correctly - ex: `{0:D}`, `{1:N}`, etc., so this does need an update to not cause an issue if formatting is used.

How far should this logic go to handle if a poorly formatted string is passed in as the Format value? The error here is really bubbling up from `string.Format(...)` when evaluating `ValueExpression` - https://github.com/dotnet/aspire/blob/ff9732fafed47140ac38226acdfef4ecd8eb5942/src/Aspire.Hosting/ApplicationModel/ReferenceExpression.cs#L49

Reply:
Modified parsing to use similar logic as is used in StringBuilder.AppendFormat (https://github.com/dotnet/runtime/blob/3393b4d564660e76178c442766f92158f04cfea2/src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs#L1518)